src/session.coffee

```diff
@@ -26,18 +27,22 @@ define [
           @shop_code = shop_code
           @flavor = flavor
           @metadata = metadata
+          @shop_cookie_policy = 'full'
 
           @_extractAnalyticsSessionOnCreate()
 
-        connect: (shop_code)->
+        connect: (shop_code, shop_cookie_policy = 'full')->
           return console?.warn?('Connect called without a shop code') unless shop_code
 
           # connect should be called only once
           return console?.warn?('Connect called multiple times') if @shop_code
 
           @shop_code = shop_code
+          @shop_cookie_policy = if shop_cookie_policy in SHOP_COOKIE_OPTIONS then shop_cookie_policy else 'full'
```

Comment:
in order to avoid this, we could replace any `== 'full'` to `!= 'basic'`

Reply:
It was the implementation I did in the past fixups. We did it with the constant in order to understand that these 2 are the acceptable cases.

Reply:
It's easier for someone to understand the possible cases